05-mvc-with-rails/rails-views-and-controllers/readme.md

```diff
@@ -1,127 +1,241 @@
+Now, in our index view, we have programmatic access to the variable `@artists`!
 
 > It should be noted that while we practice a separation of concerns we aren't just limited to querying for only artists in the artists controller. Additionally, we can store just about anything we could have normally stored in a ruby variable.
 
-In our `app/views/artists/show.html.erb`
-```html
-<h1>My favorite food is <%= @food %></h1>
-<h2><%= @artist.name %> <a href="/artists/<%= @artist.id %>/edit">(edit)</a></h2>
-<h4><%= @artist.nationality %></h4>
+Let's write some code in our view to display this data.
 
-<img class='artist-photo' src="<%= @artist.photo_url %>">
+In our `app/views/artists/index.html.erb`
+```html
+<h2>Artists <a href="/artists/new">(+)</a></h2>
 
-<h3>Songs</h3>
 <ul>
-  <% @songs.each do |song| %>
-    <!-- build the list element for each song -->
+  <% @artists.each do |artist| %>
     <li>
-      <a href="/songs/<%= song.id %>">
-        <%= song.title %> (<%= song.album %>)
+      <a href="/artists/<%= artist.id %>">
+        <%= artist.name %>
       </a>
     </li>
-
   <% end %>
 </ul>
 ```
 
-Here, in this view, you can see the use of all three instance variables we defined in the show action of the artists controller.
+Now when we vist `/artists` in the browser, we see a list (index) of all artists. We have included links to the new page, and for the show page, but we haven't created those yet so....
+
+### You-Do: Show, New, and Edit Actions (15/70)
+- Define `show`, and `new` controller actions for artists
+- Create `show`, and `new` views for artists
+- Display relevant info for an artist when you visit their show page
+- When you visit the `new` page, the relevant form should be displayed
+
+## BREAK (10/80)
+
+## WE-DO: Create Action (20/100)
+
+<!-- TODO: Write Create action code, and explain render vs redirect, and authenticity tokens and strong params.
+ -->
+
+### Render vs Redirect
+Let's take a closer look at index action of our Artists Controller:
+
+```ruby
+def index
+  @artists = Artist.all.order(:id).reverse
+end
+```
+
+If we remember back to Sinatra, you'll notice that we're not explicitly telling the application which view file to render.
+
+That's because rails has implicit rendering. Basically rails is smart enough to know if the artists controller action is called index, then it will look for the index view in the artist folder.
 
-## Break (10/65)
+You can explicitly change the implicit render by calling the `render` method in the action of a controller. Something like this:
 
-## Sanitization/Strong Params 30/95
+```ruby
+def index
+  @artists = Artist.all.order(:id).reverse
+  render :index
+end
+```
+
+You could also redirect rather than render. Who can tell me the differences between redirecting and rendering? (ST-WG)
+
+We would never do something like this, but if we did.
```

Comment:
If we never do this, why show it? There's an important difference between rendering a page after creation and redirecting the user. One allows the user to refresh the page without resubmitting the form